types/src/system/auction/bid_addr.rs

```diff
@@ -135,13 +133,8 @@ impl BidAddr {
     }
 
     /// Returns the common prefix of all delegators to the cited validator.
-    pub fn delegators_prefix(&self) -> Result<Vec<u8>, Error> {
-        let validator = self.validator_account_hash();
-        let mut ret = Vec::with_capacity(validator.serialized_length() + 2);
-        ret.push(KeyTag::BidAddr as u8);
-        ret.push(BidAddrTag::Delegator as u8);
-        validator.write_bytes(&mut ret)?;
-        Ok(ret)
+    pub fn delegators_prefix(&self) -> KeyPrefix {
+        KeyPrefix::DelegatorBidAddrsByValidator(self.validator_account_hash())
```

Comment:
I don't agree with this...later logic could map this into an enum or whatever but the fundamental behavior should be as primitive as possible (in this case, the original implementation).

Reply:
I've reverted the code changes related to auctions